### PR TITLE
Example of changing application class for different types of tests

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
@vanniktech PTAL, it's a `code` part of blog post I'm writing about mocking dependencies in different types of tests

### CODE REVIEW

**Comment by vanniktech** [comment]
:+1: implementation looks good

I also really do like the approach of changing the implementation of a dependency for various tests

**Comment by artem-zinnatullin** [comment]
Ah, got it, added `the`.

Please comment on a PR changes, because when you comment directly on a particular commit -> comment looses context when I do force push with rebased history :)

BTW, here is the article http://artemzin.com/blog/how-to-mock-dependencies-in-unit-integration-and-functional-tests-dagger-robolectric-instrumentation/, enjoy!

// Thanks for code review!

**Comment by vanniktech** [comment]
Will do next PR, thanks @artem-zinnatullin :)

File: app/src/debugUnitTests/java/com/artemzin/qualitymatters/ui/presenters/DeveloperSettingsPresenterTest.java [hunk 1]
```diff
@@ -298,7 +299,7 @@ public void changeHttpLoggingLevel_shouldChangeLevelAndNotifyView() {
 
             developerSettingsModel = mock(DeveloperSettingsModelImpl.class);
             developerSettingsView = mock(DeveloperSettingsView.class);
-            developerSettingsPresenter = new DeveloperSettingsPresenter(developerSettingsModel);
+            developerSettingsPresenter = new DeveloperSettingsPresenter(developerSettingsModel, mock(AnalyticsModel.class));
```
Comment (AlienAsRoger) [inline app/src/debugUnitTests/java/com/artemzin/qualitymatters/ui/presenters/DeveloperSettingsPresenterTest.java:302]: That doesn't really look like DI if you have `new` in you method, no?

**Pull request merged**